tests/e2e/smoke-loadbalancer/00-assert.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Shouldn't it be a statefulset?

Reply:
The OpenTelemetryCollector target allocation pod would be spawned when the CR mode is StatefulSet but the target allocation kind would be a Deployment. This KUTTL test is for testing the target allocation deployment alone.

Reply:
Could you add extra assert files, outlining what's expected in terms of deployments/statefulsets/configmaps?

Reply:
Have added more assert cases. Please let me know if these look fine.